hack/generate-values.sh

```diff
@@ -8,7 +8,7 @@ set -eu
 # Make sure bosh binary exists
 bosh --version >/dev/null
 
-VARS_FILE=/tmp/cf-vars.yaml
+VARS_FILE="$(mktemp -d)/cf-vars.yaml"
```

Comment:
I think I remember some inconsistencies across OS's when using `mktemp -d`. I've tested the following out on Mac and Linux and it has the benefit of making it more obvious where this temp dir came from:

```suggestion
VARS_FILE="$(mktemp -d -t 'cf-for-k8s.XXX')/cf-vars.yaml"
```

One side effect of using a different directory every time is that if you wanted to reuse the same certs and passwords from a previous deploy (i.e., you're upgrading instead of deleting and redeploying) you would no longer be able to simply rerun the script to generate any new needed variables. Personally, I think that's ok.

Reply:
`-t` is different in macOS and Linux, for sure. 

> One side effect of using a different directory every time is that if you wanted to reuse the same certs and passwords from a previous deploy

Maybe, we can use `$DOMAIN` and put the var store in `/tmp/${DOMAIN}/cf-vars.yml`?

Reply:
Oh yeah, I like the approach of using DOMAIN like that.  Just some bugs in the current approach that I'll make comments on directly.